src/dmd/typesem.d

```diff
@@ -3624,6 +3624,23 @@ Expression dotExp(Type mt, Scope* sc, Expression e, Identifier ident, int flag)
             }
             e = e.castTo(sc, e.type.nextOf().pointerTo());
         }
+        else if (ident == Id._tupleof)
+        {
+            if (e.op == EXP.type)
+            {
+                e.error("Can only use .tupleof on static array values (perhaps you meant `typeof(%s.init.tupleof)`?)", mt.toChars);
+                return ErrorExp.get();
+            }
+            else
+            {
+                uinteger_t length = mt.dim.toUInteger();
```

Comment:
`const length`

Reply:
const length = cast(size_t) mt.dim.toUInteger();

Also should perhaps check that the cast is not hiding some excessive value.